app/src/org/commcare/tasks/UpdateTask.java

```diff
@@ -236,6 +238,10 @@ public void setAsAutoUpdate() {
         wasTriggeredByAutoUpdate = true;
     }
 
+    public void setAsLocalUpdate() {
+        isLocalUpdate = true;
+    }
```

Comment:
I don't think I want to do it exactly like that because I was envisioning that in the future, we might use the local update functionality for something other than consumer apps, so I'm loathe to make those two things coupled. However, I agree that it makes more sense for `UpdateTask` to store an `authority` field directly, and can pass that in via the constructor instead of a separate setter method.